src/sweeps/bayes_search.py

```diff
@@ -324,12 +331,60 @@ def next_sample(
     )
 
 
+def impute(
+    goal: str,
+    metric_name: str,
+    impute_strategy: ImputeStrategy,
+    run: Optional[SweepRun] = None,
+    runs: Optional[List[SweepRun]] = None,
+) -> floating:
+    """Impute the value of a run's metric using a specified strategy."""
+    failed_val = 0.0
+    worst_func = min if goal == "maximize" else max
+    if impute_strategy == ImputeStrategy.best:
+        if run is None:
+            raise ValueError("impute_strategy == best requires a nonnull run")
+        try:
+            return run.metric_extremum(
+                metric_name, kind="minimum" if goal == "minimize" else "maximum"
+            )
+        except ValueError:
+            return failed_val
+    elif impute_strategy == ImputeStrategy.worst:
+        # we calc the max metric to put as the metric for failed runs
+        # so that our bayesian search stays away from them
+        worst_metric: floating = np.inf if goal == "maximize" else -np.inf
+        if runs is None:
+            raise ValueError("impute_strategy == worst requires nonnull list of runs")
+        for run in runs:
```

Comment:
im kinda confused why we only iterate `runs` during the `worst` case and not for the other cases

Reply:
`worst` is the worst metric across all runs in the sweep, `best` is the best metric from the target run in the sweep and `latest` is the latest metric from a target run in the sweep